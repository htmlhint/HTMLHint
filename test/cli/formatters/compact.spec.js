const ChildProcess = require('child_process')
const fs = require('fs')
const path = require('path')

describe('CLI', () => {
  describe('Formatter: compact', () => {
    it('should have stdout output with formatter compact', (done) => {
      const expected = fs
        .readFileSync(path.resolve(__dirname, 'compact.txt'), 'utf8')
        .replace(
          /\{\{path\}\}/g,
          path.resolve(__dirname, '../../html/executable.html')
        )
        .replace(/\\u001b/g, '\u001b')

      const expectedParts = expected.split('\n')

      ChildProcess.exec(
        [
          'node',
          path.resolve(__dirname, '../../../bin/htmlhint'),
          path.resolve(__dirname, '../../html/executable.html'),
          '--format',
          'compact',
        ].join(' '),
        (error, stdout, stderr) => {
          expect(typeof error).toBe('object')
          expect(error.code).toBe(1)

          expect(stdout).not.toBe('')

          const stdoutParts = stdout.split('\n')

          expect(stdoutParts.length).toBe(expectedParts.length)

          for (let i = 0; i < stdoutParts.length; i++) {
            const lineIndicator = `[L${i + 1}]: `
            expect(`${lineIndicator}${stdoutParts[i]}`).toBe(
              `${lineIndicator}${expectedParts[i]}`
            )
          }

          expect(stderr).toBe('')
          done()
        }
      )
    })
  })
})
