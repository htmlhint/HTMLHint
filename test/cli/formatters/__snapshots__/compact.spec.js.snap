// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Formatter: compact should have stdout output with formatter compact: stderr 1`] = `""`;

exports[`CLI Formatter: compact should have stdout output with formatter compact: stdout 1`] = `
"<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 8, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 8, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 8, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 9, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 9, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 9, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 10, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 10, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 10, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 11, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 11, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 11, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 12, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 12, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 12, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 13, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 13, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 13, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 14, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 14, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 14, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 15, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 15, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 15, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 16, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 16, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 16, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 17, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 17, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 17, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 18, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 18, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 18, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 19, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 19, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 19, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 20, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 20, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 20, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 21, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 21, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 21, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 22, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 22, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 22, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 23, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 23, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 23, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 24, col 7, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 24, col 14, error - The value of attribute [ bad ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 24, col 14, error - Duplicate of attribute name [ bad ] was found. (attr-no-duplication)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 25, col 22, error - Tag must be paired, no start tag: [ </input> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 26, col 3, error - Special characters must be escaped : [ < ]. (spec-char-escape)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 26, col 18, error - Special characters must be escaped : [ > ]. (spec-char-escape)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 28, col 11, error - Tag must be paired, no start tag: [ </div> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 29, col 9, error - Tag must be paired, no start tag: [ </div> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 30, col 7, error - Tag must be paired, no start tag: [ </hello> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 31, col 5, error - Tag must be paired, no start tag: [ </test> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 32, col 3, error - Tag must be paired, no start tag: [ </div> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 38, col 19, error - The value of attribute [ class ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 38, col 28, error - The value of attribute [ what ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 38, col 36, error - The value of attribute [ something ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 44, col 3, error - Tag must be paired, no start tag: [ </div> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 50, col 19, error - The value of attribute [ class ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 50, col 28, error - The value of attribute [ what ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 50, col 36, error - The value of attribute [ something ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 56, col 3, error - Tag must be paired, no start tag: [ </div> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 62, col 19, error - The value of attribute [ class ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 62, col 28, error - The value of attribute [ what ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 62, col 36, error - The value of attribute [ something ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 68, col 3, error - Tag must be paired, no start tag: [ </div> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 74, col 19, error - The value of attribute [ class ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 74, col 28, error - The value of attribute [ what ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 74, col 36, error - The value of attribute [ something ] must be in double quotes. (attr-value-double-quotes)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 80, col 3, error - Tag must be paired, no start tag: [ </div> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 81, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 82, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 83, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 84, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 85, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 86, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 87, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 88, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 89, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 90, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 91, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 92, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 93, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 94, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 95, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 96, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html: line 97, col 1, error - Tag must be paired, no start tag: [ </bad> ] (tag-pair)

[31m92 problems[39m"
`;
